osu.Game.Tests/Visual/Multiplayer/TestSceneMultiSpectatorScreen.cs

```diff
@@ -96,7 +96,7 @@ public void TestSpectatorPlayerSettingsHidden()
             AddUntilStep("all interactive elements removed", () => this.ChildrenOfType<Player>().All(p =>
                 !p.ChildrenOfType<PlayerSettingsOverlay>().Any() &&
                 !p.ChildrenOfType<HoldForMenuButton>().Any() &&
-                !p.ChildrenOfType<SongProgressBar>().Single().ShowHandle));
+                p.ChildrenOfType<SongProgressBar>().SingleOrDefault()?.ShowHandle == false));
```

Comment:
???

please check the CI again. this is hiding the test rather than correctly testing it. it only failed once out of all the runs. now it's going to silently pass on a wrong condition. i believe you need a wait step as song process is async loaded.

Reply:
yeah well, the reason I chose `AddUntilStep` was that it may take more than one frame for this check to pass due to certain components not loading at once.

will go with `AddWaitStep` then with explanatory comment

Reply:
what you had was more correct, it just read wrong to me.

Reply:
Right... my brian halted while I was pushing that change. I think `AddUntilStep` *with* an explanatory comment should read well.